lib/AST/InlinableText.cpp

```diff
@@ -189,41 +189,83 @@ struct ExtractInactiveRanges : public ASTWalker {
 };
 } // end anonymous namespace
 
+static void appendRange(
+  SourceManager &sourceMgr, SourceLoc start, SourceLoc end,
+  SmallVectorImpl<char> &scratch) {
+  unsigned bufferID = sourceMgr.findBufferContainingLoc(start);
+  unsigned offset = sourceMgr.getLocOffsetInBuffer(start, bufferID);
+  unsigned endOffset = sourceMgr.getLocOffsetInBuffer(end, bufferID);
+
+  // Strip comments from the chunk before adding it by re-lexing the range.
+  LangOptions FakeLangOpts;
+  Lexer lexer(FakeLangOpts, sourceMgr, bufferID, nullptr, LexerMode::Swift,
+    HashbangMode::Disallowed, CommentRetentionMode::ReturnAsTokens,
+    offset, endOffset);
+
+  SourceLoc nonCommentStart = start;
+  Token token;
+
+  // Re-lex the range, and skip the full text of `tok::comment` tokens.
+  while (!token.is(tok::eof)) {
+    lexer.lex(token);
+
+    if (token.is(tok::comment)) {
+      // Append the range from the last non-comment token to the beginning of this comment
+      // token.
+      SourceLoc commentLoc = token.getLoc();
+      auto charRange = CharSourceRange(sourceMgr, nonCommentStart, commentLoc);
+      StringRef text = sourceMgr.extractText(charRange);
+      scratch.append(text.begin(), text.end());
+
+      // Append a single whitespace character, to avoid fusing tokens.
+      scratch.push_back(' ');
+
+      // Set the start of the next non-comment range to the end of this token.
+      SourceLoc endLoc = Lexer::getLocForEndOfToken(sourceMgr, token.getLoc());
+
+      // The comment token's end location includes trailing whitespace, so trim trailing
+      // whitespace and only strip the portions of the comment that are not whitespace.
```

Comment:
What about leading trivia? I don’t remember how the C++ parser handles it but in SwiftSyntax newlines are leading trivia and thus we need to be careful not to remove those either. E.g. 

```swift
let a = 3
/* test */let b = 2
```

Must remain on two lines.

Reply:
I added a test for this